File: image-crud/src/main/java/org/judexmars/imagecrud/config/healthcheck/MicrometerConfig.java
package org.judexmars.imagecrud.config.healthcheck;

import io.micrometer.core.aop.CountedAspect;
import io.micrometer.core.aop.TimedAspect;
import io.micrometer.core.instrument.MeterRegistry;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.EnableAspectJAutoProxy;

/**
 * Configuration for Micrometer.
 */
@Configuration
@EnableAspectJAutoProxy
public class MicrometerConfig {

  @Bean
  CountedAspect countedAspect(MeterRegistry registry) {
    return new CountedAspect(registry);
  }

  @Bean
  TimedAspect timedAspect(MeterRegistry registry) {
    return new TimedAspect(registry);
  }
}

File: image-processor/src/main/kotlin/org/judexmars/imageprocessor/config/MicrometerConfig.kt
package org.judexmars.imageprocessor.config

import io.micrometer.core.aop.CountedAspect
import io.micrometer.core.aop.TimedAspect
import io.micrometer.core.instrument.MeterRegistry
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.EnableAspectJAutoProxy

/**
 * Configuration for Micrometer.
 */
@Configuration
@EnableAspectJAutoProxy
class MicrometerConfig {
    @Bean
    fun countedAspect(registry: MeterRegistry?): CountedAspect {
        return CountedAspect(registry!!)
    }

    @Bean
    fun timedAspect(registry: MeterRegistry?): TimedAspect {
        return TimedAspect(registry!!)
    }
}
